File: frontend/src/types/settlement.ts
export type SettlementStatus = 'pending' | 'completed'

export interface Settlement {
  id: string
  inquiry_id: string
  revenue: number
  vendor_cost: number
  other_expenses: number
  net_profit: number
  status: SettlementStatus
  notes: string | null
  created_by: string
  created_at: string
  updated_at: string
}

export interface SettlementCreate {
  inquiry_id: string
  revenue: number
  vendor_cost: number
  other_expenses?: number
  notes?: string
}

export interface SettlementUpdate {
  revenue?: number
  vendor_cost?: number
  other_expenses?: number
  notes?: string
}

export interface FnFSummary {
  total_revenue: number
  total_vendor_cost: number
  total_other_expenses: number
  total_net_profit: number
  pending_count: number
  completed_count: number
}

File: backend/app/schemas/settlement.py
import uuid
from decimal import Decimal
from pydantic import BaseModel


class SettlementCreate(BaseModel):
    inquiry_id: uuid.UUID
    revenue: Decimal
    vendor_cost: Decimal = Decimal("0")
    other_expenses: Decimal = Decimal("0")
    notes: str | None = None


class SettlementUpdate(BaseModel):
    revenue: Decimal | None = None
    vendor_cost: Decimal | None = None
    other_expenses: Decimal | None = None
    notes: str | None = None


class SettlementResponse(BaseModel):
    id: uuid.UUID
    inquiry_id: uuid.UUID
    revenue: Decimal
    vendor_cost: Decimal
    other_expenses: Decimal
    net_profit: Decimal
    status: str
    notes: str | None
    created_by: uuid.UUID
    created_at: str
    class Config:
        from_attributes = True
